<launch>

  <arg name="dataset" value="ECRot_dataset" />
  <arg name="sequence" value="bicycle" />

  <!-- Event-based Mosaicing Bundle Adjustment -->
  <node name="emba" pkg="emba" type="emba" output="screen" args="--v 0">
    <!-- dataset info -->
    <param name="dataset" value="$(arg dataset)" />
    <param name="sequence" value="$(arg sequence)" />

    <param name="dataset_root_dir" value="/home/shuang/datasets" />
    <param name="input_data_dir" value="/home/shuang/papers/emba/data" />
    <param name="output_data_dir" value="/home/shuang/papers/emba/test" />

    <!-- Topics -->
    <param name="events_topic" value="/cam0/events" />
    <param name="camera_info_topic" value="/cam0/camera_info" />

    <!-- What to refine? -->
    <param name="filename_raw_traj" value="cmaxw_traj_interp" />
    <param name="init_map_available" value="true" />
    <param name="start_time" value="0.1" />
    <param name="stop_time" value="4.9" />
    <param name="record_data" value="true" />

    <!-- Measurement model -->
    <param name="C_th" value="0.2" />

    <!-- Levenberg-Marquardt settings -->
    <param name="max_num_iter" value="50" />
    <param name="tol_fun" value="0.001" />
    <param name="num_times_tol_fun_sat" value="2" />

    <!-- Use the conjugate gradient (CG) solver? -->
    <param name="use_CG" value="false" />
    <!-- Use robust cost function (IRLS)? -->
    <param name="use_IRLS" value="false" />
    <!-- Cost funtion type: quadratic || huber || cauchy -->
    <param name="cost_type" value="quadratic" />
    <!-- Parameter for the Huber or Cauchy cost function -->
    <param name="eta" value="0.1" />

    <!-- Valid pixel selection (>= 2) -->
    <param name="thres_valid_pixel" value="5" />
    <!-- Weight of the L2 regularizer -->
    <param name="alpha" value="5.0" />
    <!-- Damping factor on the map updating -->
    <param name="damping_factor" value="1.0" />

    <!-- Mini-batch for speed-up -->
    <param name="event_batch_size" value="100" />
    <!-- Event subsampling -->
    <param name="event_sampling_rate" value="1" />

    <!-- Sliding window settings-->
    <param name="time_window_size" value="4.8" />
    <param name="sliding_window_stride" value="1.0" />
    <!-- Trajectory settings -->
    <param name="dt_knots" value="0.05" /> <!-- Linear spline -->
  </node>

</launch>
